psi4/driver/p4util/fcidump.py

```diff
@@ -0,0 +1,209 @@
+
+
+def fcidump(wfn, fname='INTDUMP', write_footer=False, oe_ints=None):
+    """Save integrals to file in FCIDUMP format as defined in Comp. Phys. Commun. 54 75 (1989)
+    Additional one-electron integrals, including orbital energies, can also be saved.
+    This latter format can be used with the HANDE QMC code but is not standard.
+
+    :returns: None
+
+    :raises: ValidationError when SCF wavefunction is neither RHF nor UHF
+
+    :type wfn: :py:class:`~psi4.core.Wavefunction`
+    :param wfn: set of molecule, basis, orbitals from which to generate cube files
+    :param fname: name of the integrals file, defaults to INTDUMP
+    :param write_footer: whether to write a timestamp in the file footer, defaults to False.
+    :param oe_ints: list of additional one-electron integrals to save to file.
+    So far only EIGENVALUES is a valid option.
+
+    :examples:
+
+    >>> # [1] Save one- and two-electron integrals to standard FCIDUMP format
+    >>> E, wfn = energy('scf', return_wfn=True)
+    >>> fcidump(wfn)
+
+    >>> # [2] Save orbital energies, one- and two-electron integrals. Print timestamp in footer.
+    >>> E, wfn = energy('scf', return_wfn=True)
+    >>> fcidump(wfn, oe_ints=['EIGENVALUES'], write_footer=True)
+
+    """
+    # Get some options
+    reference = core.get_option('SCF', 'REFERENCE')
+    ints_tolerance = core.get_option('SCF', 'INTS_TOLERANCE')
+    # Some sanity checks
+    if reference not in ['RHF', 'UHF']:
+        raise ValidationError('FCIDUMP not implemented for {} references\n'.format(reference))
+    if oe_ints is None:
+        oe_ints = []
+
+    molecule = wfn.molecule()
+    docc = wfn.doccpi()
+    frzcpi = wfn.frzcpi()
+    frzvpi = wfn.frzvpi()
+    active_docc = docc - frzcpi
+    active_socc = wfn.soccpi()
+    active_mopi = wfn.nmopi() - frzcpi - frzvpi
+
+    nbf = active_mopi.sum() if wfn.same_a_b_orbs() else 2 * active_mopi.sum()
+    nirrep = wfn.nirrep()
+    nelectron = 2 * active_docc.sum() + active_socc.sum()
+
+    core.print_out('Writing integrals in FCIDUMP format to ' + fname + '\n')
+    # Generate FCIDUMP header
+    header = '&FCI\n'
+    header += 'NORB={:d},\n'.format(nbf)
+    header += 'NELEC={:d},\n'.format(nelectron)
+    header += 'MS2={:d},\n'.format(wfn.nalpha() - wfn.nbeta())
+    header += 'UHF=.{}.,\n'.format(not wfn.same_a_b_orbs()).upper()
+    orbsym = ''
+    for h in range(active_mopi.n()):
+        for n in range(frzcpi[h], frzcpi[h] + active_mopi[h]):
+            orbsym += '{:d},'.format(h + 1)
+            if not wfn.same_a_b_orbs():
+                orbsym += '{:d},'.format(h + 1)
+    header += 'ORBSYM={}\n'.format(orbsym)
+    header += '&END\n'
+    with open(fname, 'w') as intdump:
+        intdump.write(header)
+
+    # Get an IntegralTransform object
+    check_iwl_file_from_scf_type(core.get_option('SCF', 'SCF_TYPE'), wfn)
+    spaces = [core.MOSpace.all()]
+    transType = core.IntegralTransform.TransformationType.Restricted
```

Comment:
Id like to avoid mixing camel and snake case here.

Reply:
Should I change the wrapper to the `enum`-s in `libtrans`?